nano/node/node.cpp

```diff
@@ -696,6 +695,8 @@ void nano::node::stop ()
 	epoch_upgrader.stop ();
 	workers.stop ();
 	local_block_broadcaster.stop ();
+	network.stop (); // Stop network last to avoid killing in-use sockets
```

Comment:
If we're stopping does it matter the precise time when sockets are closed?

Reply:
Ideally no, and if I was designing this code from scratch it wouldn't be needed. But for some reason the current implementation is sensitive to this, even after ensuring that channel cleanup is properly synchronised.